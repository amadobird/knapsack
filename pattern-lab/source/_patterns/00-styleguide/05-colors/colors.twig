<p>Color is a defining element of any strong brand identity. Dedicated use of the Basalt palette with reinforce the cohesiveness of the brand and visually communicates your brand personality with your audience.</p>
<p>The Basalt palette is a 5 color structure that contains two light tones, two dark tones and a brighter spot color. Together they channel natural elements while staying energetic and active creating balance and speaking to the Basalt brand story.</p>

<ul class="sg-colors">
  {% for item in items %}
    <li>
      <div class="sg-swatch" style="background: {{item.value}};"></div>
      <div class="sg-info">
        <span>{{item.value}}</span> <br/>
        <code>{{item.name}}</code>
        {% if item.comment %}<div class="comment">{{ item.comment }}</div>{% endif %}
      </div>
    </li>
  {% endfor %}
</ul>
{% if meta %}
  <p> <small>{{ meta.description }}</small> </p>
{% endif %}
