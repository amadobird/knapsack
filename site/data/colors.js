import fs from 'fs-extra';
import { join } from 'path';

/**
 * @returns {Array<Object>} - An array of colors
 */
export async function getColors() {
  const colorsFile = await fs.readFile(
    join(
      __dirname,
      '../../pattern-lab/source/_patterns/00-styleguide/05-colors/colors.json',
    ),
    'utf8',
  );
  const colors = JSON.parse(colorsFile);
  return colors.items;
}
