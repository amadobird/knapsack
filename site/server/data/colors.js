import fs from 'fs-extra';
import { join } from 'path';
import { convertColor } from '@basalt/bedrock-utils';

/**
 * @param {string} format - Color format; one of 'hsl', 'rgb', 'hex'
 * @returns {Array<Object>} - An array of colors
 */
export async function getColors(format) {
  const colorsFile = await fs.readFile(
    join(
      __dirname,
      '../../../crux-assets/_patterns/00-styleguide/05-colors/colors.json',
    ),
    'utf8',
  );
  const colors = JSON.parse(colorsFile);
  return colors.items.map(color => {
    color.value = convertColor(color.value, format);
    return color;
  });
}
