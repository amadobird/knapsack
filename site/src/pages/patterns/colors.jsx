import React from 'react';

import PatternPage from '../../templates/pattern-page';
import ColorSwatches from '../../components/color-swatch';


const ColorsPage = () => (
  <PatternPage className="docs">
    <div className="body">
      <h4 className="eyebrow">Visual Language</h4>
      <h2>Colors</h2>
      <hr />
      <blockquote>
        Mere colour, unspoiled by meaning, and unallied with definite form, can
        speak to the soul in a thousand different ways. - Oscar Wilde
      </blockquote>
      <hr />
      <p>
        Color is a defining element of any strong brand identity. Dedicated use
        of the Basalt palette with reinforce the cohesiveness of the brand and
        visually communicates your brand personality with your audience.
      </p>
      <p>
        The Basalt palette is a 5 color structure that contains two light tones,
        two dark tones and a brighter spot color. Together they channel natural
        elements while staying energetic and active creating balance and
        speaking to the Basalt brand story.
      </p>
      <p>
        Each color has a Sass variable and two utility classes. For example,
        $c-blue has : .u-c-blue for color and .u-bg-blue for background-color.
      </p>
      <ColorSwatches colors={colors} />
    </div>
  </PatternPage>
);

export default ColorsPage;
