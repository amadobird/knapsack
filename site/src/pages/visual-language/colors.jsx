import React from 'react';

import VisualLanguagePage from '../../templates/visual-language-page';
import ColorSwatches from '../../bedrock/components/color-swatch';
import { apiUrlBase } from '../../../config';
import ColorContrastBlock from '../../bedrock/components/color-contrast-block';
import ApiDemo from '../../bedrock/components/api-demo';

class ColorsPage extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      colors: [],
    };
  }

  componentDidMount() {
    window
      .fetch(`${apiUrlBase}/colors`)
      .then(res => res.json())
      .then(colors => {
        this.setState({ colors });
      });
  }

  render() {
    return (
      <VisualLanguagePage className="docs">
        <div className="body">
          <h4 className="eyebrow">Visual Language</h4>
          <h2>Colors</h2>
          <hr />
          <blockquote>
            Mere colour, unspoiled by meaning, and unallied with definite form,
            can speak to the soul in a thousand different ways. - Oscar Wilde
          </blockquote>
          <hr />
          <p>
            Color is a defining element of any strong brand identity. Dedicated
            use of the Basalt palette with reinforce the cohesiveness of the
            brand and visually communicates your brand personality with your
            audience.
          </p>
          <p>
            The Basalt palette is a 5 color structure that contains two light
            tones, two dark tones and a brighter spot color. Together they
            channel natural elements while staying energetic and active creating
            balance and speaking to the Basalt brand story.
          </p>
          <p>
            Each color has a Sass variable and two utility classes. For example,
            $c-blue has : .u-c-blue for color and .u-bg-blue for
            background-color.
          </p>
          <ColorSwatches colors={this.state.colors} />
          <ColorContrastBlock
            bgColors={this.state.colors}
            textColors={this.state.colors}
          />
          <ApiDemo
            endpoint={`${apiUrlBase}/colors`}
            queryData={{
              format: 'hsl',
            }}
            querySchema={{
              type: 'object',
              required: ['format'],
              properties: {
                format: {
                  title: 'Color Format',
                  type: 'string',
                  enum: ['hsl', 'hex', 'rgb'],
                  default: 'hsl',
                },
              },
            }}
            requestType={'POST'}
          />
        </div>
      </VisualLanguagePage>
    );
  }
}

export default ColorsPage;
