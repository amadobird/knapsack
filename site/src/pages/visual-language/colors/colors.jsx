import React from 'react';

import ColorSwatches from '@basalt/bedrock-color-swatch';
import DosAndDonts from '@basalt/bedrock-dos-and-donts';
import ColorContrastBlock from '@basalt/bedrock-color-contrast-block';
import ApiDemo from '@basalt/bedrock-api-demo';
import { BlockQuoteWrapper } from '@basalt/bedrock-atoms';
import { apiUrlBase } from '../../../../config';

class ColorsPage extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      colors: [],
    };
  }

  componentDidMount() {
    window
      .fetch(`${apiUrlBase}/colors`)
      .then(res => res.json())
      .then(colors => {
        this.setState({ colors });
      });
  }

  render() {
    return (
      <div className="docs">
        <div className="body">
          <h4 className="eyebrow">Visual Language</h4>
          <h2>Colors</h2>
          <BlockQuoteWrapper>
            Mere colour, unspoiled by meaning, and unallied with definite form,
            can speak to the soul in a thousand different ways.
            <footer>Oscar Wilde</footer>
          </BlockQuoteWrapper>
          <br />
          <p>
            Color is a defining element of any strong brand identity. Dedicated
            use of the Basalt palette with reinforce the cohesiveness of the
            brand and visually communicates your brand personality with your
            audience.
          </p>
          <p>
            The Basalt palette is a 5 color structure that contains two light
            tones, two dark tones and a brighter spot color. Together they
            channel natural elements while staying energetic and active creating
            balance and speaking to the Basalt brand story.
          </p>
          <p>
            Each color has a Sass variable and two utility classes. For example
            <code>$c-blue</code> has <code>.u-c-blue</code> for color and{' '}
            <code>.u-bg-blue</code> for background-color.
          </p>
          <br />
          <h4>Color Palette</h4>
          <p>
            The following colors make up the primary color palette of the Basalt
            brand.
          </p>
          <ColorSwatches colors={this.state.colors} />
          <br />
          <h4>Colors and Accessibility</h4>
          <p>
            Color contrast (the difference between text and background color) is
            an important aspect of accessibility standards. Text color and
            background color need to have a high enough contrast ratio to be
            readable. Test out various color combinations below to see what does
            and doesn&#39;t meet accessibility standards.
          </p>
          <ColorContrastBlock
            bgColors={this.state.colors}
            textColors={this.state.colors}
            do={this.state.do}
          />
          <ApiDemo
            title="Colors API"
            endpoint={`${apiUrlBase}/colors`}
            queryData={{
              format: 'hsl',
            }}
            querySchema={{
              type: 'object',
              required: ['format'],
              properties: {
                format: {
                  title: 'Color Format',
                  type: 'string',
                  enum: ['hsl', 'hex', 'rgb'],
                  enumNames: [
                    'HSL (Hue, Saturation, Lightness)',
                    'Hex (Hexadecimal)',
                    'RGB (Red, Green, Blue)',
                  ],
                  default: 'hsl',
                  'ui:widget': 'radio',
                },
              },
            }}
            requestType="get"
          />
        </div>
        <DosAndDonts
          items={[
            {
              image:
                '/assets/images/dos-and-donts/background-colors/background-colors-dont.png',
              caption:
                'use color combinations with low color contrast that fail WCAG standards.',
              do: false,
            },
            {
              title: 'Do Example',
              image:
                '/assets/images/dos-and-donts/background-colors/background-colors-do.png',
              caption:
                'use color combinations with distinct contrast that fulfill WCAG standards.',
              do: true,
            },
          ]}
        />
      </div>
    );
  }
}

export default ColorsPage;
