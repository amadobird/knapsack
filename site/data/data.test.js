import { getColors } from './colors';

describe('Data Get Functions', () => {
  test('Colors', async () => {
    expect(await getColors()).toMatchSnapshot();
    expect(await getColors('rgb')).toMatchSnapshot();
    expect(await getColors('hex')).toMatchSnapshot();
  });
});
