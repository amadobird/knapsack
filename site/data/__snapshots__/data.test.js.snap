// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Data Get Functions Colors 1`] = `
Array [
  Object {
    "comment": "",
    "name": "$c-blue--light",
    "value": "hsl(165, 26%, 85%)",
  },
  Object {
    "comment": "",
    "name": "$c-blue",
    "value": "hsl(200, 55%, 19%)",
  },
  Object {
    "comment": "",
    "name": "$c-green",
    "value": "hsl(159, 20%, 17%)",
  },
  Object {
    "comment": "",
    "name": "$c-iron",
    "value": "hsl(240, 3%, 85%)",
  },
  Object {
    "comment": "",
    "name": "$c-yellow",
    "value": "hsl(52, 74%, 54%)",
  },
  Object {
    "comment": "",
    "name": "$c-white",
    "value": "white",
  },
  Object {
    "comment": "",
    "name": "$c-gray--dark",
    "value": "hsl(0, 0%, 40%)",
  },
  Object {
    "comment": "",
    "name": "$c-black",
    "value": "black",
  },
]
`;

exports[`Data Get Functions Colors 2`] = `
Array [
  Object {
    "comment": "",
    "name": "$c-blue--light",
    "value": "rgb(207, 227, 222)",
  },
  Object {
    "comment": "",
    "name": "$c-blue",
    "value": "rgb(22, 57, 75)",
  },
  Object {
    "comment": "",
    "name": "$c-green",
    "value": "rgb(35, 52, 46)",
  },
  Object {
    "comment": "",
    "name": "$c-iron",
    "value": "rgb(216, 216, 218)",
  },
  Object {
    "comment": "",
    "name": "$c-yellow",
    "value": "rgb(225, 201, 51)",
  },
  Object {
    "comment": "",
    "name": "$c-white",
    "value": "rgb(255, 255, 255)",
  },
  Object {
    "comment": "",
    "name": "$c-gray--dark",
    "value": "rgb(102, 102, 102)",
  },
  Object {
    "comment": "",
    "name": "$c-black",
    "value": "rgb(0, 0, 0)",
  },
]
`;

exports[`Data Get Functions Colors 3`] = `
Array [
  Object {
    "comment": "",
    "name": "$c-blue--light",
    "value": "#CFE3DE",
  },
  Object {
    "comment": "",
    "name": "$c-blue",
    "value": "#16394B",
  },
  Object {
    "comment": "",
    "name": "$c-green",
    "value": "#23342E",
  },
  Object {
    "comment": "",
    "name": "$c-iron",
    "value": "#D8D8DA",
  },
  Object {
    "comment": "",
    "name": "$c-yellow",
    "value": "#E1C933",
  },
  Object {
    "comment": "",
    "name": "$c-white",
    "value": "#FFFFFF",
  },
  Object {
    "comment": "",
    "name": "$c-gray--dark",
    "value": "#666666",
  },
  Object {
    "comment": "",
    "name": "$c-black",
    "value": "#000000",
  },
]
`;
